packages/nuxt/src/app/components/nuxt-error-boundary.ts

```diff
@@ -11,9 +11,10 @@ export default defineComponent({
     const error = ref<Error | null>(null)
     const nuxtApp = useNuxtApp()
 
-    onErrorCaptured((err) => {
+    onErrorCaptured((err, target, info) => {
       if (process.client && !nuxtApp.isHydrating) {
         emit('error', err)
+        nuxtApp.hooks.callHook('vue:error', err, target, info)
```

Comment:
Yes, it makes sense.

The problem is that when error occurs inside an error boundary neither page:finish nor vue:error are called, so not only loading indicator remains active, but view transitions are not finished or aborted either.

Reply:
![image](https://github.com/nuxt/nuxt/assets/52774564/e2c80258-77a9-4bda-8dca-09f1ddb70f9e)

What about something like this? Or am I digging in the wrong direction?

Reply:
I think we could use `page:finish` there 👍 

The `page:` hooks are really about suspense rather than changing route. (They won't run if a page does not have an async setup or async dependencies, for example.)

Reply:
I'm pretty sure they're called even with a sync setup.

And in that case **page:finish** gets called twice with my solution :cry:
 
![image](https://github.com/nuxt/nuxt/assets/52774564/f718c975-adbb-43f7-bfb7-03cff175f012)
![image](https://github.com/nuxt/nuxt/assets/52774564/be641ac1-57a3-49d8-b592-ae7439ca7541)

Reply:
So, to recap

```js
// page.ts

onErrorCaptured(() => {
  nuxtApp.callHook('page:finish').finally(done)
})

// it works, but for sync components **page:finish** gets called twice
// because onResolve from Suspense also gets called
```